.github/workflows/go.yml

```diff
@@ -0,0 +1,40 @@
+name: Go
+on: [push]
+jobs:
+
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Set up Go 1.13
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.13
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Get dependencies
+      run: |
+        go get -v -t -d ./...
+        if [ -f Gopkg.toml ]; then
+            curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+            dep ensure
+        fi
+
+    - name: apply bin from GOPATH
+      shell: bash
+      run: echo "::add-path::$(go env GOPATH)/bin"
+      
+    - name: Build
+      run: make build
+      
+    - name: Upload artifact
+      uses: actions/upload-artifact@v1.0.0
+      with:
+        # Artifact name
+        name: output
+        # Directory containing files to upload
+        path: output
```

Comment:
We don't need artifacts for per-commit level especially for public. Please do migrate them to `on:[tag]` level since go module use tag to do a semantic versioning, and for each commit, we could build on local dev machine.

Reply:
After you migrated into `on:[tag]`, please also do a publish of release. Release notes should be a brief changelog of commit history(if possible), like this: 

```
Features:
    <Commit Short Hash>: Commit Message
Bug Fix:
    <commit Short Hash>: Commit message
```

if not possible, please just attach compiled artifacts SHA256 checksum. It's okay. Then we will manually edit relase notes.

Reply:
I will remove "upload-artifact".

Reply:
How about creating another workflow resolve release independently?

Reply:
> How about creating another workflow resolve release independently?

Of course, it should be working in that way.